src/main/java/ch/njol/skript/expressions/ExprCarryingBlockData.java

```diff
@@ -0,0 +1,88 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.ItemUtils;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.util.slot.Slot;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.entity.Enderman;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Enderman Carrying BlockData")
+@Description({
+	"The block data an enderman is carrying.",
+	"Custom attributes such as NBT or names do not transfer over."
+})
+@Examples({
+	"broadcast the carrying blockdata of last spawned enderman",
+	"set the carried block data of last spawned enderman to an oak log",
+	"set the carrying block data of {_enderman} to oak stairs[facing=north]",
+	"clear the carried blockdata of {_enderman}"
+})
+@Since("INSERT VERSION")
+public class ExprCarryingBlockData extends SimplePropertyExpression<LivingEntity, BlockData> {
+
+	static {
+		register(ExprCarryingBlockData.class, BlockData.class, "carr(ied|ying) block[ ]data", "livingentities");
+	}
+
+	@Override
+	public @Nullable BlockData convert(LivingEntity entity) {
+		if (entity instanceof Enderman enderman)
+			return enderman.getCarriedBlock();
+		return null;
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE)
+			return CollectionUtils.array(Block.class, BlockData.class, ItemStack.class, ItemType.class, Slot.class);
```

Comment:
Since you support all these types for changers, I'd recommend making a note of it in the description, considering the expression is called "... carrying **BlockData**"